riverdriver/riverpgxv5/river_pgx_v5_driver.go

```diff
@@ -29,8 +34,96 @@ type Driver struct {
+func (d *Driver) GetExecutor() riverdriver.Executor             { return &Executor{d.dbPool, dbsqlc.New()} }
+func (d *Driver) UnwrapExecutor(tx pgx.Tx) riverdriver.Executor { return &Executor{tx, dbsqlc.New()} }
+func (d *Driver) UnwrapTx(tx pgx.Tx) pgx.Tx                     { return tx }
+
+type Executor struct {
+	dbtx interface {
+		dbsqlc.DBTX
+		Begin(ctx context.Context) (pgx.Tx, error)
+	}
+	queries *dbsqlc.Queries
+}
+
+func (e *Executor) Begin(ctx context.Context) (riverdriver.ExecutorTx, error) {
+	tx, err := e.dbtx.Begin(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return &ExecutorTx{Executor: Executor{tx, e.queries}, tx: tx}, nil
+}
+
+func (e *Executor) Exec(ctx context.Context, sql string) (struct{}, error) {
+	_, err := e.dbtx.Exec(ctx, sql)
+	return struct{}{}, interpretError(err)
+}
+
+func (e *Executor) MigrationDeleteByVersionMany(ctx context.Context, versions []int) ([]*riverdriver.Migration, error) {
+	migrations, err := e.queries.RiverMigrationDeleteByVersionMany(ctx, e.dbtx, mapSlice(versions, func(v int) int64 { return int64(v) }))
+	return mapMigrations(migrations), interpretError(err)
+}
+
+func (e *Executor) MigrationGetAll(ctx context.Context) ([]*riverdriver.Migration, error) {
+	migrations, err := e.queries.RiverMigrationGetAll(ctx, e.dbtx)
+	return mapMigrations(migrations), interpretError(err)
+}
+
+func (e *Executor) MigrationInsertMany(ctx context.Context, versions []int) ([]*riverdriver.Migration, error) {
+	migrations, err := e.queries.RiverMigrationInsertMany(ctx, e.dbtx, mapSlice(versions, func(v int) int64 { return int64(v) }))
+	return mapMigrations(migrations), interpretError(err)
+}
+
+func (e *Executor) TableExists(ctx context.Context, tableName string) (bool, error) {
+	exists, err := e.queries.TableExists(ctx, e.dbtx, tableName)
+	return exists, interpretError(err)
+}
+
+type ExecutorTx struct {
+	Executor
+	tx pgx.Tx
+}
+
+func (t *ExecutorTx) Commit(ctx context.Context) error {
+	return t.tx.Commit(ctx)
+}
+
+func (t *ExecutorTx) Rollback(ctx context.Context) error {
+	return t.tx.Rollback(ctx)
 }
 
-func (d *Driver) GetDBPool() *pgxpool.Pool  { return d.dbPool }
-func (d *Driver) UnwrapTx(tx pgx.Tx) pgx.Tx { return tx }
+func interpretError(err error) error {
+	if errors.Is(err, pgx.ErrNoRows) {
+		return riverdriver.ErrNoRows
+	}
+	return err
+}
+
+func mapMigrations(migrations []*dbsqlc.RiverMigration) []*riverdriver.Migration {
+	if migrations == nil {
+		return nil
+	}
+
+	return mapSlice(migrations, func(m *dbsqlc.RiverMigration) *riverdriver.Migration {
+		return &riverdriver.Migration{
+			ID:        int(m.ID),
+			CreatedAt: m.CreatedAt,
+			Version:   int(m.Version),
+		}
+	})
+}
+
+// mapSlice manipulates a slice and transforms it to a slice of another type.
+func mapSlice[T any, R any](collection []T, mapFunc func(T) R) []R {
+	result := make([]R, len(collection))
+
+	for i, item := range collection {
+		result[i] = mapFunc(item)
+	}
+
+	return result
+}
```

Comment:
I assume this is included in the pgxv5 driver to avoid a circular dependency on `internal/util/sliceutil`?

Reply:
Yeah :/ We could add another package that both top-level `river` and the drivers could both use, but doesn't feel worth it. We could put it in `riverdriver` to share, but that feels pretty weird. I suspect this is the main function that'll be useful so there won't be too much more duplication.